frontend/public/components/deploy-image.tsx

```diff
@@ -0,0 +1,427 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { Helmet } from 'react-helmet';
+import { Link } from 'react-router-dom';
+
+import { history, Loading, NavTitle, resourcePathFromModel, Timestamp, units } from './utils';
+import { getActiveNamespace } from '../ui/ui-actions';
+import { NsDropdown } from './RBAC/bindings';
+import { k8sCreate } from '../module/k8s';
+import { ButtonBar } from './utils/button-bar';
+import { DeploymentConfigModel, ImageStreamModel, ImageStreamImportsModel, ServiceModel } from '../models';
+
+const runsAsRoot = image => {
+  const user = _.get(image, 'dockerImageMetadata.Config.User');
+  return !user ||
+          user === '0' ||
+          user === 'root';
+};
+
+const getImagePorts = image => _.get(image, 'dockerImageMetadata.Config.ExposedPorts') || _.get(image, 'image.dockerImageMetadata.ContainerConfig.ExposedPorts');
+
+const parsePorts = image => {
+  // map ports to k8s structure
+  const parsePortsFromSpec = portSpec => {
+    const ports = [];
+    _.map(portSpec, (value, key) => {
+      const parts = key.split('/');
+      if (parts.length === 1){
+        parts.push('tcp');
+      }
+
+      const containerPort = parseInt(parts[0], 10);
+      if (!_.isFinite(containerPort)) {
+        // eslint-disable-next-line no-console
+        console.log(`Container port ${parts[0]} is not a number for image ${image.metadata.name}`);
+      } else {
+        ports.push({
+          containerPort: containerPort,
+          protocol: parts[1].toUpperCase()
+        });
+      }
+    });
+
+    return _.sortBy(ports, 'containerPort');
+  };
+
+  return parsePortsFromSpec(getImagePorts(image) || []);
+};
+
+const ImagePorts = ({ports, name}) => <React.Fragment>
+  {_.size(ports) > 1 ? 'Ports ' : 'Port '}
+  {_.map(ports, (value, key) => key.toUpperCase()).join(', ')} will be load balanced by Service <strong>{name || '<name>'}</strong>.
+  <div>Other containers can access this service through the hostname <strong>{name || '<name>'}</strong>.</div>
+</React.Fragment>;
+
+export class DeployImage extends React.Component<DeployImageProps, DeployImageState> {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      namespace: '',
+      imageName: '',
+      loading: false,
+      inProgress: false,
+      name: '',
+    };
+
+    this.onNamespaceChange = this.onNamespaceChange.bind(this);
+    this.onImageNameChange = this.onImageNameChange.bind(this);
+    this.onKeyPress = this.onKeyPress.bind(this);
+    this.onNameChange = this.onNameChange.bind(this);
+    this.search = this.search.bind(this);
+    this.save = this.save.bind(this);
+  }
+
+  onNamespaceChange = (namespace: string) => {
+    this.setState({namespace});
+  };
+
+  onImageNameChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    this.setState({imageName: event.currentTarget.value});
+  }
+
+  onKeyPress = event => {
```

Comment:
You're actually binding these to this twice. The fat arrow syntax (`onKeyPress = event => { ... }`) binds to this already, but you're doing it again in the contructor. So we should pick one syntax or the other and use it consistently.

https://medium.com/@machnicki/handle-events-in-react-with-arrow-functions-ede88184bbb